### PR TITLE
Change format of Git log for how we're using it

### PR DESCRIPTION
**Pull request by dnahodil**
These messages get truncated in Slack so by changing the formatting we might be able to make the Slack messages more useful.

Screen Shot 2021-06-23 at 9 53 12 am

The main thing I'm trying to do is pull the commit message title line above the fold.

Original output
```
commit 4e0b6d7a21e19cdc0d4e5d44caca347fb11514df
Merge: 66be2e6b47 40f99d08cd
Author: jkburges <jkburges@gmail.com>
Date:   Mon Jun 21 15:28:18 2021 +1000

    Merge pull request #7503 from Biteable/ENG-362-fix-ember-deploys
    
    [ENG-362] Fix ember deploys
```

Proposed output:
```
[4e0b6d7a21] Merge pull request #7503 from Biteable/ENG-362-fix-ember-deploys
By: jkburges (jkburges@gmail.com)
Date: Mon Jun 21 15:28:18 2021 +1000

Merge pull request #7503 from Biteable/ENG-362-fix-ember-deploys
[ENG-362] Fix ember deploys
```

[ENG-362]: https://biteable.atlassian.net/browse/ENG-362
[ENG-362]: https://biteable.atlassian.net/browse/ENG-362

### CODE REVIEW

File: hooks/pre-command [hunk 1]
```diff
@@ -5,7 +5,7 @@ DIR="$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd)"
 . "$DIR/../lib/notify.bash"
 
 message() {
-  git_msg=`git log -1`
+  git_msg=`g log -1 --format="[%h] %s%nBy: %an (%ae)%nDate: %ad%n%n%s%n%b%n"`
```
Comment (wbowling) [inline hooks/pre-command:8]:
```suggestion
  git_msg=`git log -1 --format="[%h] %s%nBy: %an (%ae)%nDate: %ad%n%n%s%n%b%n"`
```

Comment (wbowling) [reply hooks/pre-command:8]: Not sure if buildkite/docker have the `g` alias

**Comment by wbowling** [comment]
Nice I like this

**Comment by jkburges** [comment]
> [4e0b6d7a21] Merge pull request #7503 from Biteable/ENG-362-fix-ember-deploys
> By: jkburges (jkburges@gmail.com)
> Date: Mon Jun 21 15:28:18 2021 +1000
> 
> Merge pull request #7503 from Biteable/ENG-362-fix-ember-deploys
> [ENG-362] Fix ember deploys

Line 5 is mostly already in line 1, can we drop it?

File: hooks/pre-command (hunk 1 above)
Comment (dnahodil) [reply hooks/pre-command]: Thanks for spotting that. Yeah, full commands ftw

**Comment by dnahodil** [comment]
> Line 5 is mostly already in line 1, can we drop it?

I'd wondered that, Jon. But I thought the body text of some commit messages might not make much sense without the first/subject line anyway. So better to be able to read them both together and I don't think space under the fold is an issue (not scarce, not pushing anything else off).

**Comment by wbowling** [comment]
Does the webapp use this plugin yet? I see there are some 1.1.0 changes that will affect the message too. Might be good to get a test setup in #deploy-testing so we can see what it looks like

**Comment by jkburges** [comment]
> Does the webapp use this plugin yet

The rails API doesn't; the pro/lite clients, renderer and transcoder all do.

**Comment by wbowling** [comment]
https://biteable.slack.com/archives/C021PLP0HRB/p1624856068000300 is a test message, new lines seem to be stripped?

image

**Comment by dnahodil** [comment]
Oh bummer.
I'll check that out

**Comment by wbowling** [comment]
Could also add it to https://github.com/Biteable/biteable-video-web/blob/master/lib/tasks/deploy/notifier.rb#L15 until rails deploy is using this plugin

**Comment by jkburges** [comment]
> until rails deploy is using this plugin

FWIW, I don't think there's anything stopping us from using the plugin in API deploy step, so feel free to try that in case it's easier.

**Comment by wbowling** [comment]
Ah https://github.com/Biteable/deploy-buildkite-plugin/pull/5/files#diff-d97ebb3562f30ce8ce601523fab4d9498d0ea5b93d405c10e4f677f7e1c4b1a8R21 needs to be quoted to preserve the new lines `echo "$msg"`

**Review by wbowling — APPROVED** [review]
lgtm :shipit: 

What's the process for a new release, just via https://github.com/Biteable/deploy-buildkite-plugin/releases/new then update anywhere that uses it?

**Comment by wbowling** [comment]
I think this will b

> What's the process for a new release, just via https://github.com/Biteable/deploy-buildkite-plugin/releases/new then update anywhere that uses it?

https://github.com/Biteable/deploy-buildkite-plugin#releasing

**Pull request merged**